.pre-commit-config.yaml

```diff
@@ -23,3 +23,14 @@ repos:
     rev: 0.23.0
     hooks:
       - id: check-github-workflows
+
+-   repo: https://github.com/pre-commit/mirrors-mypy
+    rev: v1.2.0
+    hooks:
+      - id: mypy
+        additional_dependencies: [types-PyYAML, types-setuptools, types-requests, numpy, pydantic]
```

Comment:
Update o any of these dependencies may silently break workflow and prevent any contributors from creating commits.

Reply:
If these are updated and pre-commit is run after the update, it will create a new environment with the new additional dependencies, so I don't think it would break anything?

Reply:
For example the new release of `numpy` with updated type annotation may lead to failing my py run, so fail pre-commit without bumping mypy revision. And I will pouch all active PR. When the current state pins the version of mypy and its dependencies.

Reply:
also `--check-untyped-defs` is missed from this configuration